dotCMS/src/main/java/com/dotcms/rest/StructureResource.java

```diff
@@ -65,24 +66,30 @@ public Response getStructuresWithWYSIWYGFields(@Context HttpServletRequest reque
 			beginItem = Integer.valueOf(range.substring(0, range.indexOf("-")));
 			endItem = Integer.valueOf(range.substring(range.indexOf("-") + 1));
 		}
-		
-		if(inodeFilter.isEmpty()) {
-			List<Structure> allStructures = StructureFactory.getStructures("structuretype,upper(name)", -1);
-			for(Structure st : allStructures) {
-				if(st.isArchived() == false && (nameFilter.isEmpty() || (st.getName().toLowerCase().startsWith(nameFilter)))) {
-					for(Field field : FieldsCache.getFieldsByStructureInode(st.getInode())) {
-						if(field.getFieldType().equals(Field.FieldType.WYSIWYG.toString())) {
-							structures.add(st);
-							break;
+
+		try {
+			if (inodeFilter.isEmpty()) {
+				List<Structure> allStructures = StructureFactory
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "StructureFactory"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)